smart-py/marketplace.py

```diff
@@ -53,12 +59,70 @@ def cancel_swap(self, params):
     def update_fee(self, params):
         sp.verify(sp.sender == self.data.manager)
         self.data.fee = params
-        
+
     @sp.entry_point
-    def update_manager(self, params):
+    def update_fee_account(self, params):
         sp.verify(sp.sender == self.data.manager)
+        self.data.fee_account= params
+
+    @sp.entry_point
+    def update_manager(self, params):
+        sp.if sp.now - self.data.last_ping > 100 * 24 * 3600:
+            sp.verify(self.data.secondary_accounts.contains(sp.sender))
+        sp.else:
+            sp.verify(sp.sender == self.data.manager)
+
         self.data.manager = params
-        
+        self.data.secondary_accounts.add(params)
+
+    @sp.entry_point
+    def add_secondary_account(self, params):
+        sp.verify(sp.sender == self.data.manager)
+        self.data.secondary_accounts.add(params)
+
+    @sp.entry_point
+    def remove_secondary_account(self, params):
+        sp.verify(sp.sender == self.data.manager)
+        self.data.secondary_accounts.remove(params)
+
+    @sp.entry_point
+    def ping(self):
```

Comment:
Is an explicit ping needed, or would it be enough for the manager to invoke the other methods like swap to update the last ping time?

Reply:
The swap is executed by normal users, so the last_ping time should not be updated in that case. 
I think it makes sense to have an entrypoint only for that. One could also update it in other methods that are only executed by the manager: update_fee, update_fee_account, add_secondary_account, etc

Reply:
👍